backend/canisters/group/impl/src/lib.rs

```diff
@@ -306,39 +309,32 @@ impl Data {
             pinned_messages: Vec::new(),
             test_mode,
             permissions: permissions.unwrap_or_default(),
-            invite_code: None,
-            invite_code_enabled: false,
             new_joiner_rewards: None,
             frozen: Timestamped::default(),
             timer_jobs: TimerJobs::default(),
             date_last_pinned: None,
             gate: Timestamped::new(gate, now),
+            invited_users: InvitedUsers::default(),
         }
     }
 
-    pub fn min_visible_event_index(&self, caller: Principal, invite_code: Option<u64>) -> Option<EventIndex> {
+    pub fn min_visible_event_index(&self, caller: Principal) -> Option<EventIndex> {
         match self.participants.get_by_principal(&caller) {
             Some(p) => Some(p.min_visible_event_index()),
             None => {
-                if self.is_accessible_by_non_member(invite_code) && self.history_visible_to_new_joiners {
+                if self.is_public && self.history_visible_to_new_joiners {
```

Comment:
If history is visible to new joiners then wouldn't we always return EventIndex::default()?

Reply:
Ahh I guess if it is visible then the min_visible_event_index will be set to default.